File: Frontend/src/app/core/models/recommendation.ts
export type RecommendedAction = 'Hold' | 'Reorder' | 'Share' | 'UrgentShare' | 'MarketOpportunity';

export interface Recommendation {
  id: string;
  storeId: string;
  productId: string;
  productName: string;
  batchId: string | null;
  recommendedAction: RecommendedAction;
  trendSlope: number;
  anomalyDetected: boolean;
  predictedDepletionDate: string | null;
  confidenceScore: number;
  modelVersion: string;
  reason: string;
  generatedAt: string;
  forecast: ForecastSnapshot | null;
}

export interface ForecastSnapshot {
  dates: string[];
  actuals: number[];
  yhat: number[];
  lower: number[];
  upper: number[];
}

export interface MarketSignalPoint {
  date: string;
  reservationCount: number;
  transferVolume: number;
  participatingStoreCount: number;
}

export interface RunSummary {
  productsEvaluated: number;
  recommendationsWritten: number;
  skipped: number;
  failed: number;
}

File: Domain/Entities/Recommendation.cs
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities;

public class Recommendation : BaseEntity
{
    public Guid StoreId { get; private set; }

    public Guid ProductId { get; private set; }

    public Guid? BatchId { get; private set; }

    public RecommendedAction RecommendedAction { get; private set; }

    public double TrendSlope { get; private set; }

    public bool AnomalyDetected { get; private set; }

    public DateTime? PredictedDepletionDate { get; private set; }

    public double ConfidenceScore { get; private set; }

    public string ModelVersion { get; private set; } = string.Empty;

    public string Reason { get; private set; } = string.Empty;

    public DateTime GeneratedAt { get; private set; }

    public string? ForecastSnapshotJson { get; private set; }

    private Recommendation()
    {
    }

    public Recommendation(
        Guid storeId,
        Guid productId,
        Guid? batchId,
        RecommendedAction recommendedAction,
        double trendSlope,
        bool anomalyDetected,
        DateTime? predictedDepletionDate,
        double confidenceScore,
        string modelVersion,
        string reason,
        DateTime generatedAt)
    {
        StoreId = storeId;
        ProductId = productId;
        BatchId = batchId;
        RecommendedAction = recommendedAction;
        TrendSlope = trendSlope;
        AnomalyDetected = anomalyDetected;
        PredictedDepletionDate = predictedDepletionDate;
        ConfidenceScore = confidenceScore;
        ModelVersion = modelVersion;
        Reason = reason;
        GeneratedAt = generatedAt;
    }

    public void SetForecastSnapshot(string? snapshotJson)
    {
        ForecastSnapshotJson = snapshotJson;
    }
}